### PR TITLE
feat: timestemp inside LoggerSerivce can be overridden 

### PR DESCRIPTION
**Pull request by xuzaixian**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [4262](https://github.com/nestjs/nest/issues/4262)

## What is the new behavior?

```ts
@Injectable()
class MyLogger extends LoggerService {
    getTimestamp() {
        return new Date().toISOString()
    }
}
```

then use is Logger Class
```ts
app.useLogger(app.get(MyLogger))
// or app.useLogger(new MyLogger())
```

log service prints log like 
```
[Nest] 26259   - 2020-11-10T09:38:30.098Z   [RoutesResolver] XXX {/xx}: +14ms
```
## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: packages/common/services/logger.service.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export interface LoggerService {
   warn(message: any, context?: string);
   debug?(message: any, context?: string);
   verbose?(message: any, context?: string);
+  getTimestamp?(): string;
```
Comment (kamilmysliwiec) [inline packages/common/services/logger.service.ts:16]:
```suggestion
```

I don't think we should be adding this method to the `LoggerService`. Instead, we should define an instance-level `getTimestamp()` method which will simply fall back to the static one. Hence, anyone extending the `Logger` class will be able to override it.

**Comment by kamilmysliwiec** [comment]
LGTM

**Comment by nicklasmoeller** [comment]
@kamilmysliwiec just a heads up:

This requires any logger implementors (that doesn't extend the default logger) to implement their own `getTimestamp` method, and should probably be considered a breaking change. A fallback would be appropriate if implementors don't provide that method. Especially since it's not part of the interface provided: https://github.com/nestjs/nest/blob/51e6b81bc8d51d5e62d3205b48002e88722b10a1/packages/common/services/logger.service.ts#L10-L16

The docs also document that this should be a viable path: https://docs.nestjs.com/techniques/logger#custom-implementation

**Comment by kamilmysliwiec** [comment]
Fixed in 7.6.2. Thanks for reporting @nicklasmoeller

**Pull request merged**